From 19deb667e176c9e985795efa059b4d68d6f56b21 Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin2.jansa@lgepartner.com>
Date: Fri, 23 Feb 2024 00:27:57 +0100
Subject: [PATCH] neva: Fix build with dcheck_always_on=true in GN_ARGS

:Release Notes:
DCHECK_EQ(origin_.GetOrigin(), origin_);
was added in:

  commit f9bebd4b90d6ec90e01f65d5b23b6bea2294af9e
  Author: Minsu Kim <minsu0.kim@lge.com>
  Date:   Wed Aug 23 18:18:35 2023 +0900

    [op][n_ups][build][base] Snapshot of webos-pro/chromium release changes (branch: ose/chromium108)

but there is no GetOrigin on GURL since:
https://chromium.googlesource.com/chromium/src/+/800532c0bf6712ea4ab5928da9e776d6607a10b1

:Detailed Notes:
Fixes:

:Testing Performed:
Only build tested.

:QA Notes:
No change to image.

:Issues Addressed:
[WRP-412] Create GPVB with Yocto 4.2 Mickledore

Change-Id: I902cb04d90d021433be9dc691df66a39cc76f4fe
---
Upstream-Status: Submitted [http://gpro.lge.com/c/webosose/chromium108/+/393561 neva: Fix build with dcheck_always_on=true in GN_ARGS]

 .../browser/push_messaging/push_messaging_app_identifier.cc     | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/neva/app_runtime/browser/push_messaging/push_messaging_app_identifier.cc b/src/neva/app_runtime/browser/push_messaging/push_messaging_app_identifier.cc
index a25ac56a89..c8d6c33451 100644
--- a/src/neva/app_runtime/browser/push_messaging/push_messaging_app_identifier.cc
+++ b/src/neva/app_runtime/browser/push_messaging/push_messaging_app_identifier.cc
@@ -300,7 +300,7 @@ void PushMessagingAppIdentifier::DCheckValid() const {
   DCHECK_GE(service_worker_registration_id_, 0);
 
   DCHECK(origin_.is_valid());
-  DCHECK_EQ(origin_.GetOrigin(), origin_);
+  DCHECK_EQ(origin_.DeprecatedGetOriginAsURL(), origin_);
 
   // "wp:"
   DCHECK_EQ(kPushMessagingAppIdentifierPrefix,
